pandas/tests/dtypes/cast/test_promote.py

```diff
@@ -159,12 +163,9 @@ def _check_promote(
 
     # for equal values, also check type (relevant e.g. for int vs float, resp.
     # for different datetimes and timedeltas)
-    match_value = (
+    match_value = result_fill_value == expected_fill_value and type(
         result_fill_value
-        == expected_fill_value
-        # disabled type check due to too many xfails; GH 23982/25425
-        # and type(result_fill_value) == type(expected_fill_value)
-    )
+    ) == type(expected_fill_value)
```

Comment:
I don't think I like black... That's just a terrible way to organise these two equalities.... Oh well.

Reply:
maybe pretty this up with:

```
res_type = type(result_fill_value)
ex_type = type(expected_fill_value)
match_value = result_fill_value == expected_fill_value and res_type == ex_type

Reply:
Will do.